sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/DataSourceTest.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.execution.datasources
+
+import java.io.File
+
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe.TypeTag
+
+import org.apache.spark.sql.{DataFrame, SaveMode}
+import org.apache.spark.sql.test.SQLTestUtils
+
+private[sql] trait DataSourceTest extends SQLTestUtils {
```

Comment:
Hi, @viirya . 
- The name and functions look more broader than the claim. This test suite seems to target all file-based data sources which supports Vectorization. Did I understand correctly? +1 for the idea.
- Specifically, this is used in `native` ORC, `hive` ORC and Parquet testing.

Also, cc @wangyum since he is working on Hive upgrade.

Reply:
Maybe `FileBaseDataSourceTest`?

Reply:
Sounds better than `DataSourceTest`.